### PR TITLE
feat: Change the release process

### PR DESCRIPTION
**Pull request by Erouan50**
## Description

feat: Change the way to release

Use the new release-please-action GitHub action to release the Terraform
module. This new way to release the module will open a release pull
request that will append commits from main until we're ready to create
a new release. The new way avoids manual release from the Makefile.

Also, the release process now depends on Conventional Commits where this
convention frames the way the commit message should be written. The
messages must now contain a prefix that specifies the type of the
changes (fix, feature, etc...). The GitHub action will use this prefix
to sort the commit in the changelog. It will also use it to infer the
next version number. For example, if the release contains only fixes,
the new version will be a patch (ex: v1.0.0 -> v1.0.1).

Because the repository already has a tag and release, we have to include
some bootstrap configuration otherwise release-please will use 0.1.0 as
the next release instead of 1.3.1 or 1.4.0.

To make sure commit authors respect the new convention, this commit
introduces a new pre-commit hook that enforces it.

## Type of change

- [x] New feature (non-breaking change which adds functionality)

## How Has This Been Tested?

Tests are not needed, I'm changing the release process

## Checklist:

- [x] My PR title follows [semantics prefix](.github/CONTRIBUTING.md#semantic-pull-requests)
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have checked my code and corrected any misspellings
- [x] I have run pre-commit hooks `pre-commit run -a`
- [x] CI tests are passing

### CODE REVIEW

**Review by PascalBourdier — APPROVED** [review]

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -0,0 +1,13 @@
+on:
+  push:
+    branches:
+      - master
```
Comment (elagree) [inline .github/workflows/release.yml:4]:
```suggestion
      - main
```
or change main branch to master ? :)

**Review by grandria — APPROVED** [review]

File: release-please-config.json [hunk 2]
```diff
@@ -0,0 +1,15 @@
+{
+  "bootstrap-sha": "6c543a5930c519c3fac807377b400ef423a6d312",
+  "release-type": "terraform-module",
+  "packages": {
+    ".": {
+      "release-type": "terraform-module",
+      "draft": false,
+      "prerelease": false,
+      "bumpMinorPreMajor": false,
```
Comment (xabufr) [inline release-please-config.json:9]:
```suggestion
      "bump-minor-pre-major": false,
```
And this is already the default value

File: release-please-config.json [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+      "bumpPatchForMinorPreMajor": false,
```
Comment (xabufr) [inline release-please-config.json:10]: same here

File: release-please-config.json [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+      "changelogPath": "CHANGELOG.md",
```
Comment (xabufr) [inline release-please-config.json:11]: same here

**Comment by xabufr** [comment]
many changes are not related to the way to release ?

**Comment by Erouan50** [comment]
> Some changes are not related to the way to release ? 

Damn, you're right, I rename the latest commit on `main`, I gonna rebase main on this branch.

**Pull request merged**